integration/spark/src/test/scala/org/apache/carbondata/integration/spark/testsuite/dataload/TestLoadDataGeneral.scala

```diff
@@ -310,7 +310,8 @@ class TestLoadDataGeneral extends QueryTest with BeforeAndAfterEach {
     val tableStatusFile = CarbonTablePath.getTableStatusFilePath(carbonTable.getTablePath)
     FileFactory.getCarbonFile(tableStatusFile).delete()
     sql("insert into stale values('k')")
-    checkAnswer(sql("select * from stale"), Row("k"))
+    // if table lose tablestatus file, the system should keep all data.
+    checkAnswer(sql("select * from stale"), Seq(Row("k"), Row("k")))
```

Comment:
I think this behaviour would be harrmful when the segment is marked for delete..If the segment is not removed then the results would be incorrect.

Reply:
yes
1. now, after removing the tablestatus file,  we can consider all segments are successful by default.
2. in the future, we can add a flag file into the segment when we mark the segment to be deleted.
   the flag file contains a message the segment is mark_for_delete.

Reply:
for this test case, the result is two rows,  the behavior is right.

Reply:
ok